src/Serialization/main.jl

```diff
@@ -232,7 +189,16 @@ function load_type_dispatch(s::DeserializerState, ::Type{T}, dict::Dict) where T
         return backref
     end
 
-    decodeType(dict[:type]) <: T || throw(ErrorException("Type in file doesn't match target type: $(dict[:type]) not a subtype of $T"))
+# FIXME: do we want to be covariant or contravariant in T? Or a mixture?!?
+#
+# Let U = decodeType(dict[:type])
+# use case 1:  T == U: all is fine
+# use case 2:  T is not concrete, like "Vector" (or even "Any"?); then U <: T
+# use case 3:  T is concrete, but U is not, then T <: U
+# If neither T nor U is concrete, then raise an error, perhaps? Or accept it?
```

Comment:
These comments are a bit cryptic I am afraid; I was trying to capture the rationale for the check on U a few lines down.

Perhaps we should accept more types like `Vector{Vector}` which is a subtype of `Vector` yet not concrete either...